Make some classes `GDSOFTCLASS` to avoid registering them

StarryWorm opened the pull request:
* Prepares for #106873

On `master`, when running the godot executable, multiple class registration warnings appear. 
Some of these, which this PR focuses on, are simply a matter of being made `GDSOFTCLASS` instead of `GDCLASS`. This eleminates the requirement to register them in `ClassDB`.
* Note: this can only be applied to classes which are not accessible by users, which was referenced from the latest documentation.


<details>

Make Translation Parser subclasses `GDSOFTCLASS`
Make `ResourceFormatImporterSaver` `GDSOFTCLASS` to avoid registration issues Make Godot Physics Servers `GDSOFTCLASS`
Make SceneInterfaces `GDSOFTCLASS`
Add missing editor  registrations and some to `GDSOFTCLASS` Make `SplitContainerDragger` `GDSOFTCLASS`
Make platform `IP` and `NativeMenu` implementations `GDSOFTCLASS` Make SpaceStates `GDSOFTCLASS`
Make `ProjectManager` `GDSOFTCLASS`
Make some `GLTFDocumentExtension*` classes `GDSOFTCLASS` Make `MovieWriter*` classes `GDSOFTCLASS`
</details>

dsnopek reviewed (commented):
I'm not sure about using `GDSOFTCLASS()` in all of these cases - this may require some discussion.

In particular, this feels weird for the classes that exist to implement virtual methods of a specific class, because that class will appear to be its parent via reflection. Like, if you have an instance of `EditorTheme` (maybe from an editor plugin), it will appear to be just `Theme` when inspected from GDScript. Is this what we want?

StarryWorm commented on editor/register_editor_types.cpp:
```diff
@@ -195,6 +202,15 @@ void register_editor_types() {
+	GDREGISTER_INTERNAL_CLASS(ProjectDialog);
+	GDREGISTER_INTERNAL_CLASS(ProjectList);
+	GDREGISTER_INTERNAL_CLASS(ProjectListItemControl);
+	GDREGISTER_INTERNAL_CLASS(QuickSettingsDialog);
```
Also, these changes should not be in this PR, they slipped through when I was moving commits around to the different branches. They will be moved to #111947

AThousandShips commented:
I'm not sure it's a good idea to split this off into separate PRs, any part of making #106873 work should really be in it, so that we don't do work that is only required for that PR and it ends up not being merged, was this discussed somewhere?

Is there anything that makes this PR necessary *on its own*?

StarryWorm committed:
Make some classes `GDSOFTCLASS` to avoid registering them

Make Trasnlation Parser subclasses `GDSOFTCLASS`
Make `ResourceFormatImporterSaver` `GDSOFTCLASS` to avoid registration issues
Make Godot Physics Servers `GDSOFTCLASS`
Make SceneInterfaces `GDSOFTCLASS`
Add missing editor  registrations and some to `GDSOFTCLASS`
Make `SplitContainerDragger` `GDSOFTCLASS`
Make platform `IP` and `NativeMenu` implementations `GDSOFTCLASS`
Make SpaceStates `GDSOFTCLASS`
Make `ProjectManager` `GDSOFTCLASS`
Make some `GLTFDocumentExtension*` classes `GDSOFTCLASS`
Make `MovieWriter*` classes `GDSOFTCLASS`
Make `GodotNavigationServer2D` `GDSOFTCLASS`
